### PR TITLE
feat(dashboard-LNB): set project dashboard LNB group

### PR DESCRIPTION
**Pull request by piggggggggy**
### To Reviewers
- [ ] Skip (`style`, `chore` ONLY)
- [ ] Not that difficult

### Type of Change
- [x] New feature
- [ ] Bug fixes
- [ ] Feature improvement
- [ ] Refactor
- [ ] Others (performance improvement, refactoring, CI/CD, etc.)

### Checklist
- [ ] `Error / Warning / Lint / Type`

### Description
First, just link Dashboard favorite. Not that difficult.

**Second, difficult things..**
Dashboard project LNB items have to be grouped, but that data don't have project-group data.
So, project data in reference store was imported and processed by mash-up.

 Description
 - Project LNB item may or may not have a group.
 - So, no-group items just need to be converted to LNBItem.
 - Group items should be sorted by group compared to reference store's project-id and converted to LNBItem.
 - And two data need to be merged.
 - Please check `mashUpProjectGroup` function.

image

### **Plz check files changed**

### CODE REVIEW

File: src/services/dashboards/DashboardsLNB.vue [hunk 1]
```diff
@@ -128,9 +89,81 @@ export default defineComponent({
...
+                if (state.projectItems[d.project_id]) {
+                    if (groupProjectDashboardMap[state.projectItems[d.project_id].data.groupInfo.name]) {
+                        groupProjectDashboardMap[state.projectItems[d.project_id].data.groupInfo.name].push(d);
+                    } else {
+                        groupProjectDashboardMap[state.projectItems[d.project_id].data.groupInfo.name] = [d];
+                    }
+                }
+
+                // No-group
+                noGroupList.push(d);
```
Comment (WANZARGEN) [inline src/services/dashboards/DashboardsLNB.vue:108]: suggestion: It's pretty hard to follow 🥲
  
```suggestion
                // With group
                const groupId: string|undefined = state.projectItems[d.project_id]?.data.groupInfo.name;
                if (groupId) {
                   if (groupProjectDashboardMap[groupId]) {
                      groupProjectDashboardMap[groupId].push(d);
                  } else {
                      groupProjectDashboardMap[groupId] = [d];
                  }
                }
               
                // No-group
                noGroupList.push(d);
```

File: src/services/dashboards/DashboardsLNB.vue [hunk 2]
```diff
@@ -128,9 +89,81 @@ export default defineComponent({
...
+                        to: {
+                            name: DASHBOARDS_ROUTE.DETAIL._NAME,
+                            params: {
+                                dashboardId: board.project_dashboard_id,
+                            },
+                        },
+                        favoriteType: FAVORITE_TYPE.DASHBOARD,
+                    })),
+                ]);
+            });
```
Comment (WANZARGEN) [inline src/services/dashboards/DashboardsLNB.vue:136]: suggestion: It would be nice to have a sort by project group & board name. 
The order of `Object.keys(groupProjectDashboardMap)` seems to change every time.

File: src/services/dashboards/DashboardsLNB.vue [hunk 3]
```diff
@@ -128,9 +89,81 @@ export default defineComponent({
...
+                    label: board.name,
+                    to: {
+                        name: DASHBOARDS_ROUTE.DETAIL._NAME,
+                        params: {
+                            dashboardId: board.project_dashboard_id,
+                        },
+                    },
+                    favoriteType: FAVORITE_TYPE.DASHBOARD,
+                })),
+            );
```
Comment (WANZARGEN) [inline src/services/dashboards/DashboardsLNB.vue:152]: suggestion: It would be nice to have a sort by board name.

File: src/store/modules/dashboard/type.ts [hunk 4]
```diff
@@ -45,5 +45,6 @@ export interface DomainDashboardItem extends DashboardItem {
 }
 export interface ProjectDashboardItem extends DashboardItem {
     project_dashboard_id: string;
+    project_id: string;
```
Comment (WANZARGEN) [inline src/store/modules/dashboard/type.ts:48]: question: Are these properties are the spec of backend data? I mean, `ProjectDashboardItem` is used for backend api parameter only?
If it is not, please follow the convention rule. 
We use camel case for every javascript variable, object property names.

**Review by WANZARGEN — COMMENTED** [review]
What a hard work!!
I left some comments!

File: src/services/dashboards/DashboardsLNB.vue [hunk 5]
```diff
@@ -128,9 +89,81 @@ export default defineComponent({
             ]),
         });
 
-        const filterFavoriteItems = (menuItems: LNBItem[] = []) => {
+        const mashUpProjectGroup = (dashboardList: ProjectDashboardItem[] = []): LNBMenu[] => {
+            const noGroupList = [] as ProjectDashboardItem[];
+            const groupProjectDashboardMap = {} as Record<string, ProjectDashboardItem[]>;
```
[the remark this answers is not in the document]
Comment (piggggggggy) [reply src/services/dashboards/DashboardsLNB.vue]: Reflected!

File: src/services/dashboards/DashboardsLNB.vue (hunk 1 above)
Comment (piggggggggy) [reply src/services/dashboards/DashboardsLNB.vue]: Reflected!

File: src/store/modules/dashboard/actions.ts [hunk 6]
```diff
@@ -7,7 +7,6 @@ import type { DashboardState } from '@/store/modules/dashboard/type';
 
 import ErrorHandler from '@/common/composables/error/errorHandler';
 
-// TODO: implementation
 export const loadDomainDashboard: Action<DashboardState, any> = async ({ commit }): Promise<void> => {
     try {
         const { results, total_count } = await SpaceConnector.clientV2.dashboard.domainDashboard.list({});
```
Comment (sulmoJ) [inline src/store/modules/dashboard/actions.ts:12]: There's a typo on the 26th line!

Comment (piggggggggy) [reply src/store/modules/dashboard/actions.ts:12]: Thanks!!!

**Review by WANZARGEN — APPROVED** [review]
LGTM ^^
